app/[teamId]/tasklist/TaskListWrapper.tsx

```diff
@@ -85,85 +81,91 @@ export default function TaskListWrapper({ taskList }: TaskListWrapper) {
             </div>
           </DrawerTrigger>
         ))}
-        {task && <TaskDetail task={task} />}
+        {task && (
+          <TaskDetail
+            taskListId={taskList.id}
+            task={task}
+            onClose={handleCloseDetail}
+          />
+        )}
       </Drawer>
     </div>
   );
 }
 
-function TaskDetail({ task }: { task: ITask }) {
+interface TaskDetailProps {
+  taskListId: number;
+  task: ITask;
+  onClose: () => void;
+}
+
+function TaskDetail({ task, taskListId, onClose }: TaskDetailProps) {
   const { user } = useUser();
+  const { groupId } = useGroup();
+  const { refetchById } = useTaskLists();
+  const { showSnackbar } = useSnackbar();
+  const [isEdit, setIsEdit] = useState(false);
   const { formData, handleInputChange, errorMessage, resetForm } = useForm({
-    content: '',
+    name: task.name,
+    content: task.description || '',
   });
-  const { showSnackbar } = useSnackbar();
-  const commentValid = formData.content.length > 0 && !errorMessage.content;
-  const textareaRef = useRef<HTMLTextAreaElement>(null);
-
-  const { data: comments, refetch: refetchComment } = useQuery({
-    queryKey: ['comments', task.id],
-    queryFn: () => getTaskComment({ taskId: task.id }),
-    enabled: !!task,
+  const [values, setValues] = useState({
+    name: task.name,
+    content: task.description,
   });
 
-  const sortedComments = comments?.sort(
-    (a, b) => new Date(a.createdAt).getTime() - new Date(b.createdAt).getTime(),
-  );
-
-  const { mutate: createTaskCommentMutate } = useMutation({
-    mutationFn: ({ content }: { content: string }) =>
-      createTaskComment({ taskId: task.id, content }),
-    onSuccess: () => {
-      refetchComment();
-      resetForm({ content: '' });
-      showSnackbar('댓글을 작성했습니다.');
-    },
-    onError: () => showSnackbar('댓글을 작성할 수 없습니다.', 'error'),
-  });
+  const updatedAt = format(new Date(task.updatedAt), 'yyyy.MM.dd');
+  const date = format(new Date(task.date), 'yyyy년 M월 dd일');
+  const time = format(new Date(task.date), '오후 h:mm');
 
-  const { mutate: updateTaskCommentMutate } = useMutation({
-    mutationFn: updateTaskComment,
+  const { mutate: updateTaskMutate } = useMutation({
+    mutationFn: (params: { taskId: number; body: UpdateTaskBodyParams }) =>
+      updateTask({
+        groupId: groupId,
+        taskListId: taskListId,
+        ...params,
+      }),
     onSuccess: () => {
-      refetchComment();
-      showSnackbar('댓글이 수정되었습니다.');
+      setIsEdit(false);
+      refetchById(taskListId);
+      showSnackbar('완료 여부를 변경했습니다.');
+      setValues({ ...formData });
```

Comment:
여기는 수정했을때라서 수정을 완료했습니다로 변경해주셔야할것같습니다!
```suggestion
      showSnackbar('수정을 완료 했습니다.');
```

Reply:
overflow-y-scroll만 추가했는데 가로 스크롤도 생기는군요

overflow-x-hidden 추가해서 해결했습니다!